src/evict/evict_page.c

```diff
@@ -675,63 +677,71 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
             LF_SET(WT_REC_SCRUB);
     }
 
-    /* Acquire a snapshot if coming through eviction thread route. */
+    /*
+     * Acquire a snapshot if coming through eviction thread route. Also, if we have entered eviction
+     * through application threads and we have a transaction snapshot, we will use our existing
+     * snapshot to evict pages that are not globally visible based on last_running transaction.
```

Comment:
...based on the last_running transaction...

Reply:
Fixed in 65be4b5.